pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -2615,8 +2600,147 @@ protected PersistentOfflineTopicStats internalGetBacklog(boolean authoritative)
         return quotaMap;
     }
 
+    private void internalGetBacklogSizeForNonPartitionedTopic(AsyncResponse asyncResponse, MessageIdImpl messageId, boolean authoritative) {
+        // will redirect if the topic not owned by current broker
+        validateTopicOwnership(topicName, authoritative);
+        validateTopicOperation(topicName, TopicOperation.GET_BACKLOG_SIZE);
+        if (topicName.isGlobal()) {
+            validateGlobalNamespaceOwnership(namespaceName);
+        }
+
+        PersistentTopic topic = (PersistentTopic) getTopicReference(topicName);
+        PositionImpl pos = new PositionImpl(messageId.getLedgerId(), messageId.getEntryId());
+
+        try {
+            ManagedLedgerImpl managedLedger = (ManagedLedgerImpl) topic.getManagedLedger();
+            synchronized (this) {
```

Comment:
what we process here is just ledger info in mledger, synchronizing a PersistentTopicsBase instance won't work in this case. also what data this synchronized block is trying to guard?